File: apps/api/schemas/empresas.py
from pydantic import BaseModel, EmailStr
from typing import Optional, List
from datetime import datetime

class EmpresaBase(BaseModel):
    nombre: str
    cuit: str
    direccion: Optional[str] = ""
    telefono: Optional[str] = ""
    email: EmailStr
    activa: bool = True

class CreateEmpresaRequest(EmpresaBase):
    pass

class UpdateEmpresaRequest(BaseModel):
    nombre: Optional[str] = None
    cuit: Optional[str] = None
    direccion: Optional[str] = None
    telefono: Optional[str] = None
    email: Optional[EmailStr] = None
    activa: Optional[bool] = None

    # SSO corporativo (Azure AD / Entra ID)
    ssoActive: Optional[bool] = None
    ssoDomain: Optional[str] = None
    ssoProvider: Optional[str] = None
    ssoClientId: Optional[str] = None
    ssoTenantId: Optional[str] = None
    # Secret en texto plano recibido del formulario -- se cifra antes de
    # guardar y nunca se devuelve. Si se omite o llega vacío, se conserva
    # el secret ya guardado (no se pisa).
    ssoClientSecret: Optional[str] = None

class EmpresaResponse(EmpresaBase):
    id: str
    createdAt: datetime

    ssoActive: bool = False
    ssoDomain: Optional[str] = None
    ssoProvider: Optional[str] = None
    ssoClientId: Optional[str] = None
    ssoTenantId: Optional[str] = None
    # No incluye ssoClientSecret: nunca se expone, ni cifrado.
    ssoClientSecretSet: Optional[bool] = None

    class Config:
        from_attributes = True

File: apps/web/lib/api/empresas.ts
import { api } from '../api-client';

export interface Empresa {
    id: string;
    nombre: string;
    cuit: string;
    direccion?: string;
    telefono?: string;
    email: string;
    activa: boolean;
    createdAt: string;

    // SSO corporativo (Azure AD / Entra ID)
    ssoActive?: boolean;
    ssoDomain?: string | null;
    ssoProvider?: string | null;
    ssoClientId?: string | null;
    ssoTenantId?: string | null;
    ssoClientSecret?: string | null; // solo de escritura: para guardar uno nuevo
    ssoClientSecretSet?: boolean; // informa si ya hay uno guardado, sin revelarlo
}

export const empresasApi = {
    /**
     * Listar todas las empresas (Solo SUPER_ADMIN)
     */
    async listarEmpresas(): Promise<Empresa[]> {
        return api.get('/empresas');
    },

    /**
     * Obtener detalle de una empresa
     */
    async obtenerEmpresa(id: string): Promise<Empresa> {
        return api.get(`/empresas/${id}`);
    },

    /**
     * Crear una nueva empresa
     */
    async crearEmpresa(data: Partial<Empresa>): Promise<Empresa> {
        return api.post('/empresas', data);
    },

    /**
     * Actualizar una empresa
     */
    async actualizarEmpresa(id: string, data: Partial<Empresa>): Promise<Empresa> {
        return api.put(`/empresas/${id}`, data);
    },

    /**
     * Eliminar o desactivar una empresa
     */
    async eliminarEmpresa(id: string): Promise<{ message: string }> {
        return api.delete(`/empresas/${id}`);
    },
};
